core/trino-main/src/main/java/io/trino/sql/planner/planprinter/HashingAnonymizer.java

```diff
@@ -0,0 +1,356 @@
+    }
+
+    @Override
+    public String anonymize(TableHandle tableHandle)
+    {
+        return anonymize(tableHandle.getCatalogName().getCatalogName(), ObjectType.CATALOG);
+    }
+
+    @Override
+    public String anonymize(TableExecuteHandle tableHandle)
+    {
+        return anonymize(tableHandle.getCatalogName().getCatalogName(), ObjectType.CATALOG);
+    }
+
+    private String anonymize(CreateTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getHandle().getCatalogName().getCatalogName(),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(InsertTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getHandle().getCatalogName().getCatalogName(),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(RefreshMaterializedViewTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getInsertHandle().getCatalogName().getCatalogName(),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(DeleteTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getHandle().map(TableHandle::getCatalogName).map(CatalogName::getCatalogName).orElse(""),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(UpdateTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getHandle().map(TableHandle::getCatalogName).map(CatalogName::getCatalogName).orElse(""),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(TableExecuteTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getExecuteHandle().getCatalogName().getCatalogName(),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private <T> String anonymize(T object, ObjectType objectType)
+    {
+        String hashCode = Hashing.murmur3_128().hashString(environment + object.toString(), UTF_8).toString();
```

Comment:
`long hashCode = Hashing.murmur3_128().hashUnencodedChars(environment + object).asLong();`

Reply:
Why not `toString()` which returns hex string? `asLong()` could return negative values.

Reply:
We can use `Hashing.murmur3_128().hashUnencodedChars(environment + object).toString()` then